Source/Core/Core/AchievementManager.cpp

```diff
@@ -145,6 +150,7 @@ void AchievementManager::LoadGameByFilenameAsync(const std::string& iso_path,
     }
     ActivateDeactivateLeaderboards();
     ActivateDeactivateRichPresence();
+    m_do_frame_event = AfterFrameEvent::Register([this] { DoFrame(); }, "AchievementManager");
```

Comment:
Is AfterFrameEvent appropriate for this case? This could be a potential gotcha for cheevos/lboards that rely on the rate this fires ([not necessarily ~50/60 "fps"](https://github.com/dolphin-emu/dolphin/blob/ee28d332b040d07d3d9a093d0950c63abaf928c0/Source/Core/VideoCommon/BPStructs.cpp#L346-L350))

Reply:
Not sure I understand the gotcha here, is this something we should discuss within RetroAchievements?

Reply:
Simple once you realize several lbs would be counting frames, and thus the amount of times this is triggered. You might say expect 60 frames be 1 second for some game which has that in the VPS counter. Except for some games this is just going to be called more often than that (120 times perhaps), so what you think is 1 second is actually half a second. Purely an example anyways, it could go theoretically any direction. Also, saying 60 here is possibly just wrong, due to the game lagging maybe and thus not sending frames that fast, again leading to a potential gotcha. Maybe the game is sending frames much slower intentionally. Also, the fact this is a game controlled rate (the game can decide to throw a 1000 frames a second if it wants to, or decide to stop sending frames) and not a constantish rate like Vertical Blanks (what might be more appropriate?)

Reply:
I can understand what you mean in terms of measuring time durations, but that sounds like something that we'd want to account for at the developer level. If you're suggesting that, if the game were to suddenly attempt 1000 frames per second, that there should be frames we skip, that sounds just as potentially harmful to our achievement logic as calling do_frame each of those thousand times.

Reply:
`Vertical Blanks (what might be more appropriate?)`

There aren't many examples to pick out from, but N64 would be relatively close in the frames/vblank relationship, and going off the existing N64 emulators with RetroAchievements, VBlanks are used to represent frames for rcheevo purposes rather than the more "modern"ish concept of a frame.

Reply:
Taking a look through the available events; would AfterPresent be appropriate for this?

Reply:
BeforePresent/AfterPresent presents the same problem. Even more problematic as that behavior differs between Immediately Present XFB being enabled or disabled.

Callback_NewField might be more appropriate (called every VBlank and usually around 50/59.94 times a second, although the game can slightly deviate from this (although you'd know that just based on the VPS counter, likely not problematic).